src/compiler/program.ts

```diff
@@ -782,6 +782,24 @@ namespace ts {
             resolveTypeReferenceDirectiveNamesWorker = (typeReferenceDirectiveNames, containingFile, redirectedReference) => loadWithLocalCache<ResolvedTypeReferenceDirective>(Debug.checkEachDefined(typeReferenceDirectiveNames), containingFile, redirectedReference, loader);
         }
 
+        const actualResolveModuleNamesWorker = resolveModuleNamesWorker;

```

Comment:
Personally, I like the pattern where things are mutable during initialization and immutable thereafter, because I think immutable things are easier to reason about.  To do that here, I'd rename the original `let resolveModuleNamesWorker` to `let actualResolveModuleNamesWorker` and then declare `const resolveModuleNamesWorker` here, so that users are accessing and closing over a constant.  I'm fine with this version though.